### PR TITLE
Use `jsonptr` to represent paths

### PR DESCRIPTION
**Pull request by asmello**
RFC 6902 references [RFC 6901](https://datatracker.ietf.org/doc/html/rfc6901) where it chooses to define the paths in the patch operations as JSON Pointers. JSON Pointers can be encoded as strings, but they are conceptually a separate type, with very particular set of invariants that have to be maintained. This motivates defining a separate Rust type to represent them, as was done in the [`jsonptr`](https://github.com/chanced/jsonptr) crate, so that manipulations over these paths can be performed safely and with reusable code.

While use of `String`s is sufficient to implement the required JSON Patch logic in this crate with minimal boilerplate, it makes it inconvenient for crate users to manipulate the patch operations outside of the `diff` and `merge` APIs. For example, a manual implementation of `merge` for a type other than a `serde_json::Value` would have to either implement its own string manipulation utilities for extracting tokens from the paths for traversal, or convert to a `jsonptr::Pointer`, which requires a clone. The reverse may also be required if the user crate intends to produce its own JSON Patches.

By embracing the `jsonptr::Pointer` type as the representation for the operation paths in this crate, not only we make things simpler and more efficient at the consumer side, we also avoid implementing some error-prone logic in this crate, as it can take advantage of the useful APIs provided by the `jsonptr` crate. The resulting code ends up being simpler, more elegant and easier to reason about on both sides.

In short, why reinvent the wheel? If there's a perfectly good representation for JSON Pointers in the Rust open-source ecosystem, may as well converge on it.

# Side changes
* Adding a `PatchOperation::path` method here for user convenience, since every operation has this field and it's useful for directing the operation to its appropriate location without need for matching explicitly on the operation variant.
* Changed the error message of PatchError to start with a lowercase character (as [is conventional in std](https://doc.rust-lang.org/std/error/trait.Error.html), and to match the `jsonptr` messages),

### CODE REVIEW

**Comment by asmello** [comment]
Wow that was quick! Thanks @idubrov
